docs/connector-development/tutorials/custom-python-connector/4-check-and-error-handling.md

```diff
@@ -0,0 +1,67 @@
+# Check and error handling
+In this section, we'll implement the check operation, and implement error handling to surface the user-friendly messages when failing due to authentication errors.
+
+Let's first implement the check operation.
+
+This operation verifies that the input configuration supplied by the user can be used to connect to the underlying data source.
+
+Use the following command to run the check operation:
+```
+poetry run source-survey-monkey-demo check --config secrets/config.json
+```
+
+The command succeed, but it'll succeed even if the config is invalid. We should modify the check so it fails if the connector is unable to pull any record a stream.
+
+We'll do this by trying to read a single record from the stream, and fail the connector could not read any.
+```python
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
```

Comment:
I feel like the template should provide a sensible implementation by default instead of returning `True, None`. it's pretty useless